### PR TITLE
Add rerank model type for LocalAI provider

### LINKED ISSUE #3524: Extend API support for LocalAI
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Pleas do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing?

image
Dify recognizes LocalAI's LLM and embedding features only currently. But LocalAI supports many other features as well. See below screenshot.
image
It supports `TTS`, `Speach-2-text`, `text-2-image` and `image-2-image`.

### 2. Describe the feature you'd like to see

Please allow choose LocalAI as the backend for `TTS`, `STT`, `text-2-image` and `image-to-image` features.

### 3. How will this feature improve your workflow or experience?

I use LocalAI as the all-in-one LLM backend for vary models, extends the support for LocalAI simplifies the maintenance work.

### 4. Additional context or comments

_No response_

### 5. Can you help us with this feature?

- [X] I am interested in contributing to this feature.

### PR DESCRIPTION
**Pull request by thiner**
# Description

This PR extends API support for LocalAI backend. Below API will be added:

- rerank

This PR fixes #3524  

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

First draft version, haven't gotten resources to test it.

- [ ] TODO

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods
- [ ] `optional` I have made corresponding changes to the documentation 
- [ ] `optional` I have added tests that prove my fix is effective or that my feature works
- [ ] `optional` New and existing unit tests pass locally with my changes

### CODE REVIEW

File: web/yarn.lock
Comment (takatost) [inline web/yarn.lock:1]: Please remove irrelevant content. 😁

**Review by takatost — CHANGES_REQUESTED** [review]

File: web/yarn.lock
Comment (thiner) [reply web/yarn.lock:1]: reverted

**Review by thiner — COMMENTED** [review]
lgtm

**Review by takatost — APPROVED** [review]
LGTM

**Pull request merged**